stdlib/public/core/AnyHashable.swift

```diff
@@ -189,11 +198,23 @@ public struct AnyHashable {
 
 extension AnyHashable: Equatable {
   /// Returns a Boolean value indicating whether two type-erased hashable
-  /// instances wrap the same type and value.
+  /// instances wrap the same value of equivalent type.
   ///
-  /// Two instances of `AnyHashable` compare as equal if and only if the
-  /// underlying types have the same conformance to the `Equatable` protocol
-  /// and the underlying values compare as equal.
+  /// `AnyHashable` considers bridged counterparts (such as a `String` and an
+  /// `NSString`) of the same value to be equivalent when type-erased. Where
+  /// those compatible types have different definitions of equality comparisons,
+  /// values that originally compare as not equal may then compare as equal when
+  /// they are type-erased by conversion to `AnyHashable`:
```

Comment:
Suggested edits:

> `AnyHashable` considers bridged counterparts (such as a `String` and an `NSString`) of the same value to be equivalent when type-erased. Where those compatible types **use** different definitions **for equality, values that were originally distinct may become equal when they are converted** to `AnyHashable`:

Reply:
The style for writing reference avoids using type names as common nouns.  Apple Style prefers "if" for conditional text, not "where"; also "might" for possibility to avoid "may" which can also indicate permission.

> `AnyHashable` considers bridged counterparts (such as `String` and `NSString`) that have the same value to be equivalent when type-erased. If those compatible types use different definitions for equality, values that were originally distinct might compare as equal when they are converted to AnyHashable: